src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -1018,6 +1018,54 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
     return convertMavlinkWaypointsToCockpit(missionItems)
   }
 
+  /**
+   * Set home waypoint on the vehicle
+   * @param { [number, number] } coordinates Coordinates of the home waypoint
+   * @param { number } altitude Altitude of the home waypoint
+   */
+  async setHomeWaypoint(coordinates: [number, number], altitude: number): Promise<void> {
+    await this.sendCommandLong(MavCmd.MAV_CMD_DO_SET_HOME, 0, 0, 0, 0, coordinates[0], coordinates[1], altitude)
+  }
+
+  /**
+   * Fetch the home waypoint from the vehicle
+   * and converts the message into a Waypoint.
+   * @returns {Promise<Waypoint>} The home waypoint.
+   */
+  async fetchHomeWaypoint(): Promise<Waypoint> {
+    await this.sendCommandLong(MavCmd.MAV_CMD_GET_HOME_POSITION)
+    const startTime = new Date().getTime()
+    let homePosition: Message.HomePosition | undefined = undefined
+    while (!homePosition && new Date().getTime() - startTime < 5000) {
```

Comment:
Does Cockpit have an abstraction for the [MAVLink command protocol](https://mavlink.io/en/services/command.html)? This should be waiting for the [`COMMAND_ACK`](https://mavlink.io/en/messages/common.html#COMMAND_ACK), so it can give a reason if there's a failure/rejection.

Reply:
It does, indeed, line 142.

Reply:
As noticed below, the `sendCommandLong` actually uses the `sendCommand` function, so it does have the ACK already.